packages/react/src/components/Panel/index.test.tsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Panel, { PanelHeader, PanelContent } from './';
+import axe from '../../axe';
+
+test('should render the default heading level', () => {
+  render(<Panel heading={'Title'}>Content</Panel>);
+
+  expect(screen.getByRole('heading', { name: 'Title' }).tagName).toBe('H2');
+  expect(screen.getByText('Content')).toBeInTheDocument();
+});
+
+test('should render a custom heading level', () => {
+  render(<Panel heading={{ text: 'Title', level: 3 }}>Content</Panel>);
+
+  expect(screen.getByRole('heading', { name: 'Title' }).tagName).toBe('H3');
+  expect(screen.getByText('Content')).toBeInTheDocument();
+});
+
+test('should render the label the panel by the heading by default', () => {
+  render(<Panel heading={'Title'}>Content</Panel>);
+
+  const panel = screen.getByRole('region');
+  const heading = screen.getByRole('heading', { name: 'Title' });
+  const body = screen.getByText('Content');
+
+  const headingId = heading.getAttribute('id');
+  const ariaLabelledBy = panel.getAttribute('aria-labelledby');
+
+  expect(heading.tagName).toBe('H2');
+  expect(ariaLabelledBy).toBe(headingId);
+  expect(body).toBeInTheDocument();
```

Comment:
I realize this was in the original test, but some of these things are redundant since they're tested above. I'm also scratching my head on the test name here, because I think it's saying "should label the panel by the heading by default".

What I think may need to be tested here is validating that the panel has an accessible name (`screen.queryByRole('region').hasAccessibleName(...)`)

Reply:
I took a look at my work and couldn't remember why I wrote it, so I decided to revise it to make more sense.